compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -207,8 +208,19 @@ impl<Prov> Allocation<Prov> {
         align: Align,
         mutability: Mutability,
     ) -> Self {
-        let bytes = Box::<[u8]>::from(slice.into());
-        let size = Size::from_bytes(bytes.len());
+        let slice: Cow<'a, [u8]> = slice.into();
+        let size = Size::from_bytes(slice.len());
+        let align_usize: usize = align.bytes().try_into().unwrap();
+        let layout = std::alloc::Layout::from_size_align(slice.len(), align_usize).unwrap();
+        let bytes = unsafe {
+            let buf = std::alloc::alloc(layout);
+            let mut boxed = Box::<[MaybeUninit<u8>]>::from_raw(std::slice::from_raw_parts_mut(buf as *mut MaybeUninit<u8>, slice.len()));
```

Comment:
I think this is not valid because `from_raw_parts_mut` creates a mutable _reference_ to uninitialized memory.
`std::ptr::slice_from_raw_parts_mut` should work here.

Reply:
Nevermind. I missed the cast to MaybeUninit.